lib/Text_Wiki/Text/Wiki/Parse/Default/Newline.php

```diff
@@ -48,7 +48,11 @@ class Text_Wiki_Parse_Newline extends Text_Wiki_Parse {
     * @see parse()
     *
     */
-    public $regex = '/([^\n])\n(?!\n)/m';
+    public $regex = '/
+        ([^\n])  # Match any character that is not a newline
+        \n       # Then a newline
+        (?!\n)   # ...so long as it is not followed by another newline
+        /mx';
```

Comment:
I think this is how it groups newlines and requires `@@@@` for long newline segments